src/System.Windows.Forms/src/System/Windows/Forms/Dialogs/CommonDialogs/FolderBrowserDialog.cs

```diff
@@ -373,13 +398,34 @@ private void SetOption(FILEOPENDIALOGOPTIONS option, bool value)
 
     private unsafe void GetResult(IFileOpenDialog* dialog)
     {
-        using ComScope<IShellItem> item = new(null);
-        dialog->GetResult(item);
-        if (!item.IsNull)
+        if (Multiselect)
+        {
+            using ComScope<IShellItemArray> itemArray = new(null);
+            dialog->GetResults(itemArray);
+            itemArray.Value->GetCount(out uint itemCount);
+            List<string> tempSelectedPaths = new();
+            for (uint itemIndex = 0; itemIndex < itemCount; itemIndex++)
+            {
+                using ComScope<IShellItem> item = new(null);
+                itemArray.Value->GetItemAt(itemIndex, item);
+                if (!item.IsNull)
+                {
+                    item.Value->GetDisplayName(SIGDN.SIGDN_FILESYSPATH, out PWSTR ppszName);
+                    tempSelectedPaths.Add(ppszName.ToStringAndCoTaskMemFree());
```

Comment:
Looks build is complaining about this. `ppszName` 's value should not be null if `GetDisplayName` succeeded according to [docs](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ishellitem-getdisplayname). We can fix the build error doing the following:
```suggestion
                    // An exception would've been received if GetDisplayName failed.
                    tempSelectedPaths.Add(ppszName.ToStringAndCoTaskMemFree()!);
```

Reply:
I was just checking into it myself and was just going to ask if we should suppress the null-check, thanks for confirming it's safe.